src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs

```diff
@@ -91,9 +91,9 @@ public void ValidAccessLevelCombinations(MemoryMappedFileAccess mapAccess, Memor
                 }
                 catch (UnauthorizedAccessException)
                 {
-                    if (PlatformDetection.IsInContainer && (viewAccess == MemoryMappedFileAccess.ReadExecute || viewAccess == MemoryMappedFileAccess.ReadWriteExecute))
+                    if ((RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || PlatformDetection.IsInContainer) && (viewAccess == MemoryMappedFileAccess.ReadExecute || viewAccess == MemoryMappedFileAccess.ReadWriteExecute))
```

Comment:
Can you add a comment to the code about SIP?  Otherwise it may not be clear why this condition exists.  Thanks.